api/lib/opentelemetry/distributed_context/propagation/trace_parent.rb

```diff
@@ -0,0 +1,124 @@
+# frozen_string_literal: true
+
+# Copyright 2019 OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+module OpenTelemetry
+  module DistributedContext
+    module Propagation
+      # A TraceParent is an implementation of the W3C trace context specification
+      # https://www.w3.org/TR/trace-context/
+      # {SpanContext}
+      class TraceParent
```

Comment:
The layout of this class is not what I am used to seeing. I'm not sure if we have a convention, or if we should have one, but typically I'm used to seeing something like:

```ruby
class MyClass
  #constants
  MY_CONST = 1
  class << self
     #public class methods
     def public_class_method; end
     
     private
     
    #private class methods
     def private_class_method; end
  end
  
  #public instance methods

  def public_instance_method; end

  private
  
  #private instance methods
  def private_instance_method; end
end
```

Would there be any objections to adopting a similar format for OTel Ruby (and this file)?

Reply:
so you want the "static" methods up top?  That doesn't bother me at all, I'll move it up.

Reply:
Yeah, there are some that are in the `class << self` block and a few that are `self.` prefixed. If you wouldn't mind moving them into one `class << self` block at the top that'd be awesome!

Reply:
Looks great!